src/adr/0020-cassandra-mailbox-object-consistency.md

```diff
@@ -0,0 +1,71 @@
+# 20. Cassandra Mailbox object consistency
+
+Date: 2020-02-27
+
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+Mailboxes are denormalized in Cassandra in order to access them both by their immutable identifier and their mailbox 
+path (name):
+
+ - `mailbox` table stores mailboxes by their immutable identifier
+ - `mailboxPathV2` table stores mailboxes by their mailbox path
+
+We furthermore maintain two invariants on top of these tables:
+ - **mailboxPath** unicity. Each mailbox path can be used maximum once. This is ensured by writing the mailbox path first
+ using Lightweight Transactions.
+ - **mailboxId** unicity. Each mailbox identifier is used by only a single path. We have no real way to ensure a given mailbox
+ is not referenced by two paths.
+
+Failures during the denormalization process will lead to inconsistencies between the two tables.
+
+This can lead to the following user experience:
+
+```
+BOB creates mailbox A
+Denormalization fails and an error is returned to A
+
+BOB retries mailbox A creation
+BOB is being told mailbox A already exists
+
+BOB tries to access mailbox A
+BOB is being told mailbox A does not exists
+```
+
+## Decision
+
+We should provide an offline webadmin task to solve mailbox object inconsistencies.
```

Comment:
please define offline. as far as I know you cannot run any task with a stopped James